### PR TITLE
New vmc cloud provider

### PR DESCRIPTION
**Pull request by agonzalezrh**

### CODE REVIEW

**Review by ahsen-shah — APPROVED** [review]

File: ansible/cloud_providers/vmc_destroy_env.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  hosts: localhost
+  connection: local
+  gather_facts: false
+  become: false
+  tasks:
+    - name: Install community.vmware collection for vmware modules
+      command: >-
+        ansible-galaxy collection install
+        -p {{ playbook_dir }}/../collections community.vmware
+        --force-with-deps
```
Comment (fridim) [inline ansible/cloud_providers/vmc_destroy_env.yml:14]: mhh  we should probably have this setup differently.
Requirement files should be used with pinned version.

requirements-vmc.yml?

**Review by fridim — CHANGES_REQUESTED** [review]
Why do you remove ansible/configs/sandbox/infra.yml?

Do you delete all the dns record  you created?

File: ansible/cloud_providers/vmc_infrastructure_deployment.yml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    - infra_tasks
+  tasks:
+    - name: Install community.vmware collection for vmware modules
+      command: >-
+        ansible-galaxy collection install
+        -p collections community.vmware:3.1.0
+        --force-with-deps
+      args:
+        chdir: "{{ playbook_dir }}/.."
+        creates: "{{ playbook_dir }}/../collections/ansible_collections/community/vmware"
```
[the remark this answers is not in the document]
Comment (agonzalezrh) [reply ansible/cloud_providers/vmc_infrastructure_deployment.yml]: could you check the new approach?

File: ansible/roles-infra/infra-dns/defaults/main.yml [hunk 3]
```diff
@@ -9,4 +9,6 @@ infra_dns_inventory_var: >-
   r_osp_facts
   {%- elif cloud_provider == 'equinix_metal' -%}
   r_equinix_metal_devices
+  {%- elif cloud_provider == 'vmc' -%}
+  hostvars
```
[the remark this answers is not in the document]
Comment (agonzalezrh) [reply ansible/roles-infra/infra-dns/defaults/main.yml:13]: It is more complex, the public IPs are on NSX-T and not on VMware, so I get the public ips first

https://github.com/redhat-cop/agnosticd/blob/vsphere/ansible/roles-infra/infra-dns/tasks/nested_loop.yml#L10

**Pull request closed without merging**